src/test.rs

```diff
@@ -1262,8 +1417,11 @@ cargo:rerun-if-env-changed=SYSTEM_DEPS_BUILD_INTERNAL
 cargo:rerun-if-env-changed=SYSTEM_DEPS_TESTLIBWITHRPATH_BUILD_INTERNAL
 cargo:rerun-if-env-changed=SYSTEM_DEPS_TESTDATA_BUILD_INTERNAL
 cargo:rerun-if-env-changed=SYSTEM_DEPS_TESTDATA_LINK
+cargo:rerun-if-env-changed=SYSTEM_DEPS_TESTDATA_MODIFIER
 cargo:rerun-if-env-changed=SYSTEM_DEPS_TESTLIBWITHRPATH_LINK
+cargo:rerun-if-env-changed=SYSTEM_DEPS_TESTLIBWITHRPATH_MODIFIER
 cargo:rerun-if-env-changed=SYSTEM_DEPS_LINK
+cargo:rerun-if-env-changed=SYSTEM_DEPS_MODIFIER
```

Comment:
What's the rationale to have a global env var as well as the lib one?

Reply:
My motivation here was the personal use case I had 3 libraries and all of them needed the `+whole-archive` modifier to compile our test cases (due to this change: https://github.com/rust-lang/rust/pull/128400). The global here made sense to me but I can remove if we decide this is something we won't use frequently (or we can add it back later).

Reply:
> I can remove if we decide this is something we won't use frequently (or we can add it back later)

Yes, I'd rather not committing to a global API for now. I'm still not clear if this is something library specific or library user specific so let's keep this out for now.

Reply:
Of course I'll make that change as soon as I can!

Reply:
Fixed in dfff668